torchao/dtypes/nf4tensor.py

```diff
@@ -501,3 +509,9 @@ def linear_nf4(input: torch.Tensor, weight: NF4Tensor) -> torch.Tensor:
         weight: NF4Tensor weight
     """
     return LinearNF4.apply(input, weight)
+
+def to_nf4(tensor,
+           block_size: int = 64,
+           scaler_block_size: int = 256):
+    tensor1 = tensor.to(torch.bfloat16)
```

Comment:
Would vote for raising error if input is not of bf16 dtype, as this downcast could be unexpected for users. Will let you and @drisspg make the call though

Reply:
I think this makes sense, tbh, I don't think I made any assumptions in the code that require the original weigh to bfloat16, and this was motivated by checkpoint weights being in bf16.

Also I I think if this is the main entrypoint for users to create nf4 tensors then we should document this function